### PR TITLE
arch/x86: add x2APIC support

### PR DESCRIPTION
**Pull request by ghost**
Basically, add support for CONFIG_X2APIC option and modifications to the local APIC interrupt driver and local APIC timer driver to honor the option, with clean up along the way.  Some of that clean up (which may seem unrelated, but is, at least peripherally) is the relocation and renaming of MSR definitions/accessors to their own header, simplification of legacy i8259 handling, and the removal of the x86 jailhouse target.

Note that the local APIC timer driver is in pretty poor condition.  It's not been maintained properly for some time, and it needs serious attention.  That is not addressed in this patch set.

This code is tested on real hardware, and on the qemu_x86 target, though no test is included here (yet). A simple strategy to cover these changes would be to enable CONFIG_X2APIC on tests/kernel/sleep for x86 arch only; this requires some trivial changes to CI (qemu: +x2apic) which is why the test is lacking.

### CODE REVIEW

**Review by tyrantlizard77 — APPROVED** [review]

File: arch/x86/core/crt0.S [hunk 1]
```diff
@@ -415,6 +415,12 @@ __csSet:
 1:
 #endif
 
+#if defined(CONFIG_PIC_DISABLE)		/* disable i8259s if present */
+	movb $0xff, %al
+	outb %al, $0x21
```
Comment (nashif) [inline arch/x86/core/crt0.S]: just add come comment about what is being done here and where these numbers came from, for example:
```

    /* Disable PIC command */
   	movb $0xff, %al
    /* Port 2, master base address (0x20) */
    outb %al, $0x21
    /* Port 2, slave base address (0xa0) */
	outb %al, $0xA1
```

File: drivers/interrupt_controller/loapic_intr.c [hunk 2]
```diff
@@ -93,8 +94,24 @@ static int loapic_init(struct device *unused)
 	ARG_UNUSED(unused);
 	s32_t loApicMaxLvt; /* local APIC Max LVT */
 
-	/* enable the Local APIC */
-	x86_write_loapic(LOAPIC_SVR, x86_read_loapic(LOAPIC_SVR) | LOAPIC_ENABLE);
+	/*
+	 * enable the local APIC. note that we use xAPIC mode here, since
```
Comment (nashif) [inline drivers/interrupt_controller/loapic_intr.c]: indent...

File: drivers/interrupt_controller/loapic_intr.c [hunk 3]
```diff
@@ -93,8 +94,24 @@ static int loapic_init(struct device *unused)
...
+	 * x2APIC access is not enabled until the next step (if at all).
+	 */
+
+	x86_write_xapic(LOAPIC_SVR, x86_read_xapic(LOAPIC_SVR) | LOAPIC_ENABLE);
+
+#ifdef CONFIG_X2APIC
+	/*
+	 * turn on x2APIC mode. we trust the config option, so
+	 * we don't check CPUID to see if x2APIC is supported.
```
Comment (nashif) [inline drivers/interrupt_controller/loapic_intr.c]: indent

Comment (gnuless) [reply drivers/interrupt_controller/loapic_intr.c]: @nashif Not quite sure what you're looking for here.. line 102 was exactly 80 characters, which is a bit ugly, so I've broken that up.  Is that what you meant?

File: drivers/interrupt_controller/loapic_intr.c (hunk 2 above)
Comment (gnuless) [reply drivers/interrupt_controller/loapic_intr.c]: @nashif Not sure what you mean ...

File: arch/x86/core/crt0.S (hunk 1 above)
Comment (gnuless) [reply arch/x86/core/crt0.S]: @nashif I've added some explanatory comments.

File: drivers/interrupt_controller/loapic_intr.c (hunk 2 above)
Comment (nashif) [reply drivers/interrupt_controller/loapic_intr.c]: ignore, my browser is showing wrong indentation.

File: drivers/interrupt_controller/loapic_intr.c (hunk 3 above)
Comment (nashif) [reply drivers/interrupt_controller/loapic_intr.c]: ignore, my browser is showing wrong indentation.

**Review by nashif — APPROVED** [review]

**Pull request merged**